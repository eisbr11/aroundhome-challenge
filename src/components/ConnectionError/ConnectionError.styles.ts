import { createUseStyles } from 'react-jss';

const useStyles = createUseStyles({
  wrapper: {
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    flex: 1,
  },
  errorMessage: {
    color: 'red',
    fontWeight: 700,
    textAlign: 'center',
  },
});

export default useStyles;
