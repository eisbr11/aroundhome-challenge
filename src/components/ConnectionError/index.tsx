import Typography from 'components/Typography';
import useStyles from './ConnectionError.styles';

/**
 * Connection Error Message Component
 *
 * @returns {JSX}
 * @constructor
 */
const ConnectionError = () => {
  const classes = useStyles();
  return (
    <div className={classes.wrapper}>
      <Typography className={classes.errorMessage}>
        Es konnte keine Verbindung zum Server hergestellt werden.
      </Typography>
    </div>
  );
};

export default ConnectionError;
