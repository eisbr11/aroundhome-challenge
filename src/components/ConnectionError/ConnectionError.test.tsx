import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import ConnectionError from './index';

describe('Connection Error Component', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<ConnectionError />);

    expect(wrapper.length).toBe(1);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
