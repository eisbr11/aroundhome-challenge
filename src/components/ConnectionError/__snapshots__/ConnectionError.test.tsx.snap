// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connection Error Component should render correctly 1`] = `
<div
  className="wrapper-0-2-1"
>
  <Typography
    className="errorMessage-0-2-2"
  >
    Es konnte keine Verbindung zum Server hergestellt werden.
  </Typography>
</div>
`;
